src/accelerate/commands/launch.py

```diff
@@ -168,11 +184,100 @@ def tpu_launcher(args):
         xmp.spawn(main_function, args=(), nprocs=args.num_processes)
 
 
-def sagemaker_launcher(sagemaker_config, args):
-    raise NotImplementedError(
-        "Support for starting SageMaker training is not yet implemented. But you can create configs for it."
+def _convert_nargs_to_dict(nargs: List[str]) -> Dict[str, str]:
+    if len(nargs) < 0:
+        return {}
+    # helper function to infer type for argsparser
+
+    def _infer_type(s):
+        try:
+            s = float(s)
+
+            if s // 1 == s:
+                return int(s)
+            return s
+        except ValueError:
+            return s
+
+    parser = argparse.ArgumentParser()
+    _, unknown = parser.parse_known_args(nargs)
+    for index, argument in enumerate(unknown):
+        if argument.startswith(("-", "--")):
+            action = None
+            if index + 1 < len(unknown):  # checks if next index would be in list
+                if unknown[index + 1].startswith(("-", "--")):  # checks if next element is an key
+                    action = "store_true"
+            else:  # action_true parameter is last element
+                action = "store_true"
```

Comment:
There should be something for `store_false` too.

Reply:
puh. I that even possible? like how can someone cloud tell the difference between `store_true` and `store_false` without knowing the script...

Reply:
I have no idea what you're doing it here then. Also I though the script could not have `--store_true`/`--store_false`

On top of the documentation, we should add a nice error message here if the parser has arguments with `--store_true`/`--store_false`.

Reply:
For SageMaker to work I needed to convert the `args.training_script_args` from `--foo bar --john 2 --store_true --another key` to
 ```python
{
"foo":"bar",
"john": 2,
"store_true": True,
"another": "key"
}
```
To achieve this I built this dynamic args parser, which takes the `args.training_script_args` and creates an `argsparser` out of it to apply afterward on `args.training_script_args` to create this `dict`. 

While creating this I haven't thought of `store_false`, since I never have seen it in real use, that's why I expected that there is only `store_true` and implemented it like that.

Reply:
As discussed I adjusted the convert script to raise an error if someone uses `store` actions and added a case into the test.